### PR TITLE
Beginner javascript notes

### PR DESCRIPTION
**Pull request by Asjas**
Hey Wes, 

I'm creating this PR to contain the Beginner JavaScript Notes as I'm working on it. That way I can push incremental changes to it and you'll be able to review it 🙂 

I finished the content for Module 1. I also created the Table of Contents and a Aside section that contains the headings per page.

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -100,6 +100,56 @@ async function makeTipsFromMdx({ graphql, actions }) {
...
+    throw new Error('There was an error');
+  }
+  const javascriptPosts = data.allMdx.edges;
+
+  javascriptPosts.forEach((post, i) => {
+    const prev = javascriptPosts[i - 1];
+    const next = javascriptPosts[i + 1];
+
+    actions.createPage({
+      path: `/beginner-javascript-notes${post.node.fields.slug}`,
```
Comment (wesbos) [inline gatsby-node.js:140]: can you make it /javscript?

File: gatsby-node.js [hunk 2]
```diff
@@ -141,6 +191,14 @@ exports.createPages = async ({ graphql, actions }) => {
   await Promise.all([
     makePostsFromMdx({ graphql, actions }),
     makeTipsFromMdx({ graphql, actions }),
+    makeJavaScriptFromMdx({ graphql, actions }),
+    paginate({
+      graphql,
+      actions,
+      collection: 'javascript',
+      pathPrefix: '/beginner-javascript-notes/',
```
Comment (wesbos) [inline gatsby-node.js:199]: /javascript here as well

File: src/javascript/module-01-the-basics/11-types-booleans-and-equality/11-types-booleans-and-equality.mdx [hunk 3]
```diff
@@ -0,0 +1,111 @@
+---
+tocTitle: 11 - Types - Booleans and Equality
+title: Types - Booleans and Equality
+slug: module-01-the-basics/types-booleans-and-equality
```
Comment (wesbos) [inline src/javascript/module-01-the-basics/11-types-booleans-and-equality/11-types-booleans-and-equality.mdx:4]: could we take `module-` out and have the url just be `01-the-basics/...`

**Comment by wesbos** [comment]
Looks awesome! A few minor things in the reivew.

Last thing, could we put the sub-headings underanth the module in the sidebar? Just a massive nested UL/OL? I can work the styling to make sense once its all in there

**Comment by Asjas** [comment]
Yeah, I'll make the changes and push it 👍

**Comment by wesbos** [comment]
https://deploy-preview-152--wesbos.netlify.app/beginner-javascript-notes/module-01-the-basics/running-and-loading-js 

awesome! Exactly what I was envisioning!

**Comment by wesbos** [comment]
can you remove the dates from the post as well? This stuff is evergreen so it won't need a timestamp

File: src/javascript/module-01-the-basics/11-types-booleans-and-equality/11-types-booleans-and-equality.mdx (hunk 3 above)
Comment (Asjas) [reply src/javascript/module-01-the-basics/11-types-booleans-and-equality/11-types-booleans-and-equality.mdx]: `module-` has been removed from the slugs 👍

**Comment by wesbos** [comment]
Looks great! I've started work on the preliminary CSS. 

One thing either of us can tackle - whoever gets to it first -is that the sidebar should scroll to the current page when you load a page.

Also i want to use the settings for gatsby-link to show the currennt page and style it somehow

**Comment by wesbos** [comment]
just pushing my updates but the internet is slow, so ill check back tomorrow to see if it finished

**Comment by wesbos** [comment]
Just pushed some mroe updates. I think all that is left to do is the style the side by side view, and style the initial table of contents view. Almost there!

**Pull request closed without merging**